client/tx/tx.go

```diff
@@ -158,7 +158,16 @@ func SignWithPrivKey(
 	var sigV2 signing.SignatureV2
 
 	// Generate the bytes to be signed.
-	signBytes, err := txConfig.SignModeHandler().GetSignBytes(ctx, signMode, signerData, txBuilder.GetTx())
+	var (
+		signBytes []byte
+		err       error
+	)
+	h, ok := txConfig.SignModeHandler().(authsigning.SignModeHandlerWithContext)
+	if ok {
+		signBytes, err = h.GetSignBytesWithContext(ctx, signMode, signerData, txBuilder.GetTx())
+	} else {
+		signBytes, err = txConfig.SignModeHandler().GetSignBytes(signMode, signerData, txBuilder.GetTx())
+	}
```

Comment:
Rather than duplicate this code, why don't we create a helper function in `signing`:
```go
func GetSignBytesWithContext(h SignModeHandler, ctx context.Context, mode signing.SignMode, data SignerData, tx sdk.Tx) ([]byte, error) {
  hWithCtx, ok := h.(SignModeHandlerWithContext)
  if ok {
 		return hWithCtx.GetSignBytesWithContext(ctx, signMode, signerData, txBuilder.GetTx())
 	} else {
 		return h.GetSignBytes(signMode, signerData, txBuilder.GetTx())
 	}
}